src/gov/nist/javax/sip/stack/SIPDialog.java

```diff
@@ -4409,8 +4409,13 @@ public void setEarlyDialogTimeoutSeconds(int seconds) {
     public void checkRetransmissionForForking(SIPResponse response) {
         final int statusCode = response.getStatusCode();
         final String responseMethod = response.getCSeqHeader().getMethod();
-        final long responseCSeqNumber = response.getCSeq().getSeqNumber();   
-        boolean isRetransmission = !responsesReceivedInForkingCase.add(statusCode + "/" + responseCSeqNumber + "/" + responseMethod);            
+        final long responseCSeqNumber = response.getCSeq().getSeqNumber();
+        RSeq rseq = (RSeq) response.getHeader(RSeqHeader.NAME);
+        String responseRSeqNumber = "";
+        if(rseq != null) {
+            responseRSeqNumber = "/" + rseq.getSeqNumber();
+        }
+        boolean isRetransmission = !responsesReceivedInForkingCase.add(statusCode + "/" + responseCSeqNumber + "/" + responseMethod + responseRSeqNumber);
```

Comment:
Your conclusion is right. If the is no RSeq present (reliable transport is not used) the code behaves previously.
Yes this issue was discovered using service specific unit test and the problem disappeared after the fix was implemented. This was the only test we have.
Regarding the optional configuration within that it won't be needed as the additional logic indeed works only for prov resps when they are required. Making responses filtering or implementing configuration will bring some obsolete processing overhead.